src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexEmbeddedLanguage.cs

```diff
@@ -1,39 +1,50 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification.Classifiers;
 using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
-using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
 using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices
 {
-    internal sealed class RegexEmbeddedLanguage : IEmbeddedLanguage
+    internal class RegexEmbeddedLanguage : IEmbeddedLanguage
     {
-        public int StringLiteralKind { get; }
-        public ISyntaxFactsService SyntaxFacts { get; }
-        public ISemanticFactsService SemanticFacts { get; }
-        public IVirtualCharService VirtualCharService { get; }
-
-        public IEmbeddedBraceMatcher BraceMatcher { get; }
-        public IEmbeddedClassifier Classifier { get; }
-        public IEmbeddedHighlighter Highlighter { get; }
-        public IEmbeddedDiagnosticAnalyzer DiagnosticAnalyzer { get; }
-        public IEmbeddedCodeFixProvider CodeFixProvider { get; }
-
-        public RegexEmbeddedLanguage(
-            int stringLiteralKind,
-            ISyntaxFactsService syntaxFacts,
-            ISemanticFactsService semanticFacts,
-            IVirtualCharService virtualCharService)
+        public readonly EmbeddedLanguageInfo Info;
+
+        public ISyntaxClassifier Classifier { get; }
+
+        public RegexEmbeddedLanguage(EmbeddedLanguageInfo info)
+        {
+            this.Info = info;
```

Comment:
>this. [](start = 12, length = 5)

nit: inconsistent use of `this.` on Info and Classifier.

Reply:
fixed.